### ISSUE TITLE
Extra space after <xref> to <paragraphs>

### ISSUE THREAD

**Issue by Jahrme** [rbeezer/mathbook#381]
The non-breaking space added after titles in xrefs should be conditionally added. In cases where the referenced object *does not* have a number, there would be no non-breaking space added. In cases where the referenced object *does* have a number, the non-breaking space would be added as it is currently.

**Comment by rbeezer** [comment]
Good catch.  Code is too big of a mess to reliably tell which situation we are in (`prefix` is sometimes actually the whole clickable).  #385 will track issues to be addressed.  Thanks.
